### ISSUE TITLE
Complications with preloading fallback fonts

### ISSUE THREAD

**Issue by kumarasinghe** [brillout/vite-plugin-ssr#618]
### Description

Firstly, thank you for making this brilliant framework!

In short, I'm hosting a font needed for my project locally and importing it in CSS:

```css
/* PageShell.scss */
@font-face {
  font-family: Roboto;
  font-display: swap;
  src: local("Roboto"),
    url("../../fonts/Roboto-Regular.woff2") format("woff2"),
    url("../../fonts/Roboto-Regular.ttf") format("ttf");    /* fallback font */
}

```

A `ttf` fallback font source is specified above, in case the browser doesn't support `woff2`. There are two issues I want to highlight.

1. `writeEarlyHints` API fails when using fallback fonts in **production**:

```js
Server running at http://localhost:3000
TypeError [ERR_INVALID_ARG_VALUE]: The argument 'hints' must be an array or string of format "</styles.css>; rel=preload; as=style". Received '</assets/Roboto-Regular.b01691f5.woff2>; rel=preload; as=font; crossorigin'
    at new NodeError (node:internal/errors:393:5)
    at validateLinkHeaderFormat (node:internal/validators:455:11)
    at validateLinkHeaderValue (node:internal/validators:481:7)
    at ServerResponse.writeEarlyHints (node:_http_server:314:16)
```
2. This fallback font gets injected into the `<head>` element, and then loaded **even if it's is not needed**:

```html
<head>
  ...
  <link rel="preload" href="/assets/Roboto-Regular.b01691f5.woff2" as="font" type="font/woff2" crossorigin="">
  <link rel="preload" href="/assets/Roboto-Regular.4e147ab6.ttf" as="font" type="font/ttf" crossorigin="">
  ....
</head>
```

This unnecessary resource load will bring down page performance (FCP) if the developer defines multiple fonts.
I believe the default preload strategy should be updated to omit fallback font sources.

Thank you!

Related: [https://vite-plugin-ssr.com/preload](https://vite-plugin-ssr.com/preload)

**Comment by brillout** [comment]
Can you create a reproduction for `1.`?

Good idea/point  regarding `2.`.

**Comment by brillout** [comment]
> reproduction

Adding a font to the app generated by `$ npm init vite-plugin-ssr@latest` should do the trick.

**Comment by brillout** [comment]
As per Node.js's thrown error, it seems like Early Hints don't have a `crossorigin` property.

```
TypeError [ERR_INVALID_ARG_VALUE]: The argument 'hints' must be an array or string of format
"</styles.css>; rel=preload; as=style". Received '</assets/Roboto-Regular.b01691f5.woff2>;
rel=preload; as=font; crossorigin'`
```

**Comment by brillout** [comment]
Released in `0.4.74` (both the fix and the fallback enhancement https://github.com/brillout/vite-plugin-ssr/issues/624).

In case your company is up for it: https://github.com/sponsors/brillout.
